app/src/main/java/com/movies/presentation/home/ui/HomeFragment.kt

```diff
@@ -43,6 +36,21 @@ class HomeFragment : BaseFragment<HomeUIState, HomeViewModel>() {
         }
     }
 
+    private lateinit var moviePagingAdapter: MoviePagingAdapter
+
+    override fun onBind() {
+        handleResult(DETAILS) {
+            viewModel.onCreate()
+        }
+        handleResult(FAVOURITES) {
+            viewModel.onCreate()
+        }
+        initRecyclerView()
+        setListeners()
+        binding.moviesRecyclerView.adapter = moviePagingAdapter
```

Comment:
move adapter line inside the initRecyclerView

Reply:
it's a lateinit property and it should be initialized in onBind. Otherwise it will crash

Reply:
yes, but it crashed when I moved adapter line in the function :D